### PR TITLE
Adds Add Post Functionality and integrates Darkrai Chat

### PR DESCRIPTION
**Pull request by NeelMakadiya14**
Added Post Functionality in Addpost.js. 
Used Image picker to select an image and then sent post request to Cloudinary API and then sent the response URL to the server.
Improve the UI by providing the material UI using react-native-paper.
Did some changes in App.js and change the logic of rendering of sign In and Home screen. According to previous logic, the Loading screen is accessing the state variable before the rendering of the screen and throwing the warning. So, I removed that screen and used useEffect in App.js to retrieve the credential and then based on that render the Sign In or Home screen.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -74,13 +91,66 @@ export default function App() {
...
+      });
+      AsyncStorage.getItem("nickname",(err,name)=>{
+        dispatch({type:"SetNickname",name});
+      })
+    } 
+    else{
+      dispatch({type:"LOGIN",user:null,token:null});
+    }
+  });
+    },1000)
```
Comment (hitgo00) [inline App.js:131]: Try to use async/await or promises instead of setTimeout and callbacks @NeelMakadiya14

**Review by hitgo00 — COMMENTED** [review]
Good work @NeelMakadiya14 💯🎯

File: App.js (hunk 1 above)
Comment (NeelMakadiya14) [reply App.js]: Okay.
Did those changes.
Please Check.

**Comment by hitgo00** [comment]
Solves #3 and #4

**Comment by hitgo00** [comment]
Great work @NeelMakadiya14

**Comment by hitgo00** [comment]
give @NeelMakadiya14 150 points

**Pull request merged**